frontend/components.js

```diff
@@ -4,6 +4,239 @@ import actions from './actions'
 import translations from './translations'
 import {RefreshIcon, ExitIcon} from './svg'
 import printAda from './printAda'
+import {importWalletSecret, isWalletExportEncrypted} from '../wallet/keypass-json'
+
+class LoadKeyFileClass extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fileName: ' ',
+      keyFile: undefined,
+    }
+    this.selectFile = this.selectFile.bind(this)
+    this.readFile = this.readFile.bind(this)
+    this.dragOver = this.dragOver.bind(this)
+    this.drop = this.drop.bind(this)
+    this.unlockKeyfile = this.unlockKeyfile.bind(this)
+    this.updatePassword = this.updatePassword.bind(this)
+    this.closePasswordModal = this.closePasswordModal.bind(this)
+  }
+
+  closePasswordModal() {
+    this.setState({encrypted: undefined, error: undefined})
+  }
+
+  updatePassword(e) {
+    this.setState({password: e.target.value})
+  }
+
+  unlockKeyfile() {
+    this.props.loadingAction('Unlocking key file')
+    setTimeout(
+      async () =>
+        await importWalletSecret(this.state.keyFile, this.state.password)
+          .then((sec) => {
+            this.setState({error: undefined})
+            this.props.loadWallet({cryptoProvider: 'mnemonic', secret: sec.toString('hex')})
+          })
+          .catch((e) => {
+            this.props.stopLoadingAction()
+            this.setState({error: 'Wrong password'})
+          }),
+      100
+    )
+  }
+
+  selectFile(e) {
+    this.props.loadingAction('Reading file')
+    const file = e.target.files[0]
+    e.target.value = null
+    setTimeout(() => this.readFile(file), 100)
+  }
+
+  drop(e) {
+    e.stopPropagation()
+    e.preventDefault()
+    this.props.loadingAction('Reading file')
+    const file = e.dataTransfer.files[0]
+    setTimeout(() => this.readFile(file), 100)
+  }
+
+  dragOver(e) {
+    e.stopPropagation()
+    e.preventDefault()
+    e.dataTransfer.dropEffect = 'copy'
+  }
+
+  async readFile(file) {
+    this.setState({
+      fileName: file.name,
+      keyFile: undefined,
+      error: undefined,
+    })
+
+    const reader = new FileReader()
+    await reader.readAsText(file)
+
+    reader.onload = ((theFile) => {
+      return async (e) => {
+        try {
+          const json = await JSON.parse(e.target.result)
```

Comment:
"json" name for variable is imho too generic. Maybe walletExport or something along those lines would be more suitable.